test/fakeintake/client/client_integration_test.go

```diff
@@ -0,0 +1,44 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package client
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/server"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIntegrationClient(t *testing.T) {
```

Comment:
True fact: it helped me spot a bug - I was querying `fake` instead of `fakeintake` initially. I can skip it when running on the CI, if it risks to add flakiness or if it increases the CI time.

Reply:
Isn't it something you can test with `httptest` unit tests?

Reply:
[httptest.Server](https://pkg.go.dev/net/http/httptest#NewServer) does not support route definitions - I can test the request path. But in my case I wouldn't have spotted it without an integration test, as it was an integration bug. What are your concerns for keeping it ?

Reply:
It just that having the 2 kinds of test appears redundant. When someone is coming to this code in 6 months and see the different options, it raises question of why different types and what should I do?

There's no difference between unit tests and integration tests in terms of run time, environment or coverage. In that case we might as well test everything with the actual server.

Reply:
There is a difference in coverage: integration tests ensure that client and server both work correctly together. Unit test ensure that each one works well in isolation. I don't test the same use cases in integrations and unit tests. for instance, it is easier to test bad response handling with unit test. 

I see how the good path test is fairly similar in coverage, I can remove this one keeping only the integration test, but I'd rather keep both as they are slightly different: in unit test I can ensure that the server properly receives parameters as I expect to send them,  I can go deeper in the assertion, I have more control.

About readability of tests, I can add comments to describe why we have both unit and integration tests.